Meshtastic/Helpers/BLEManager.swift

```diff
@@ -11,6 +11,7 @@ import OSLog
 // Meshtastic BLE Device Manager
 // ---------------------------------------------------------------------------------------
 class BLEManager: NSObject, CBPeripheralDelegate, MqttClientProxyManagerDelegate, ObservableObject {
+	static var shared: BLEManager! // Singleton instance
```

Comment:
I have some long-term testability concerns about making BLEManager a Singleton.

Could we use @Environment and dependency injection to the places where it's used instead?

Reply:
@72A12F4E IMO something needs to be done about BLEManager being transient though. For instance, currently the Share Location functionality is capable of spinning up multiple timers during consecutive connect / disconnect cycles because of the lifecycle of that BLEManager object. 
To repro: Take a device with location sharing on the default interval, connect and disconnect multiple times in succession.

Reply:
Perhaps the timer stuff needs to be up an abstraction layer? Not sure what the idiomatic way of doing this is